### PR TITLE
Roch/multiparticle pt fluct

### LINKED ISSUE #251: Implement a general Jackknife sampler class
This could be useful for cases, where the computation of uncertainties is more complicated with the usual ways.

I've already started the implementation of such a class, since I need this for the estimation of the multi particle transverse momentum correlations (#156).

I started with a first implementation in the `Utilities.py` file, but realized that the class grew quite fast. This is why I moved it to the `src/sparkx` in a separate file. In principle, the users will be able to use this class on any other data array they provide.

### PR DESCRIPTION
**Pull request by Hendrik1704**
This adds:
- Multi particle $\langle p_{\rm T}\rangle$ correlations and cumulants up to order 8 with uncertainties
- Delete-d Jackknife class for statistical Jackknife uncertainty computation (parallelized with Pool)
- Corresponding class documentations and unit tests
- Sort documentation index in alphabetical order
- All new classes and functions implement the statical typing (issue #247)

@NGoetz If you can think of any other missing tests, feel free to add them. This was all I could come up with at this point. For the physics implementation, I haven't found a good test yet.

This closes #251 and #156.

### CODE REVIEW

File: src/sparkx/MultiParticlePtCorrelations.py [hunk 1]
```diff
@@ -0,0 +1,500 @@
...
+
+        Parameters
+        ----------
+        particle_list_all_events : list
+            List of events, where each event is a list of particle objects.
+        """
+        for event in particle_list_all_events:
+            self._transverse_momentum_correlations_event_num_denom(event)
+
+    def _compute_mean_pt_correlations(self, data: np.ndarray) -> float:
```
Comment (NGoetz) [inline src/sparkx/MultiParticlePtCorrelations.py:238]: I think the name data is here a bit unspecific.

File: src/sparkx/MultiParticlePtCorrelations.py [hunk 2]
```diff
@@ -0,0 +1,500 @@
...
+
+        kappa_store = np.zeros(self.max_order)
+        kappa_store_error = np.zeros(self.max_order)
+        for order in range(self.max_order):
+            dataN = self.N_events[:, :2*(order+1)]
+            dataD = self.D_events[:, :2*(order+1)]
+            # combine the numerator and denominator for each order in data
+            # always alternate the columns of the numerator and denominator
+            combined_data = np.empty((dataN.shape[0], dataN.shape[1] + dataD.shape[1]), dtype=dataN.dtype)
+            combined_data[:, 0::2] = dataN
```
Comment (NGoetz) [inline src/sparkx/MultiParticlePtCorrelations.py:487]: I have to say that this confused me a tiny bit in the beginning. I think we can keep it like this, but the next time I would just use two arguments.

**Review by NGoetz — APPROVED** [review]
Looks all good and reasonable to me. Have you used it already? Does it produce reasonable results? I am still a bit unsure about physics test, as I do not have intuition for this. How would it look if all particles have the same pt?

File: src/sparkx/MultiParticlePtCorrelations.py (hunk 1 above)
Comment (Hendrik1704) [reply src/sparkx/MultiParticlePtCorrelations.py:238]: I changed the name to `numerator_denominator_array`

File: src/sparkx/MultiParticlePtCorrelations.py (hunk 2 above)
Comment (Hendrik1704) [reply src/sparkx/MultiParticlePtCorrelations.py:487]: I agree, it took a few minutes to come up with this.

**Review by NGoetz — APPROVED** [review]
Looks good, thank you for the additional test.

**Pull request merged**